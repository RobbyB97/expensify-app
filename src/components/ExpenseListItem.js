import React from 'react'
import {Link} from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'

const ExpenseListItem = ({id, description, amount, createdAt}) => (
  <div>
    <h6>{description}</h6>
    <p>
        {numeral(amount/100).format('$0,0.00')} 
        | 
        {moment(createdAt).format('MMM DD YYYY')}
    </p>
    
    <Link to={`/edit/${id}`}>
        Edit Expense
    </Link>

  </div>
)

export default ExpenseListItem
