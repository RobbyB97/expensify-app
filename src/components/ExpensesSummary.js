import React from 'react'
import {connect} from 'react-redux'
import numeral from 'numeral'

import expensesTotal from '../selectors/expensesTotal'


export const ExpensesSummary = (props) => (
    <div>
        {
            props.expensesCount ?
                `You have ${props.expensesCount} expenses totalling ${props.expensesTotal}.`
            :
            'You have no expenses.'
        }
    </div>
)


const mapStateToProps = (state) => {
    return {
        expensesTotal: numeral(expensesTotal(state.expenses)/100).format('$0,0.00'),
        expensesCount:  state.expenses.length
    }
}


export default connect(mapStateToProps)(ExpensesSummary)