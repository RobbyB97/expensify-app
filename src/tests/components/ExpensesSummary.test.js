import React from 'react'
import {shallow} from 'enzyme'
import {ExpensesSummary} from '../../components/ExpensesSummary'


test('Render ExpensesSummary with no expenses', () => {
    const wrapper = shallow(<ExpensesSummary />)
    
    expect(wrapper).toMatchSnapshot()
})


test('Render ExpensesSummary with one expense', () => {
    const wrapper = shallow(
        <ExpensesSummary 
            expensesCount={1}
            expensesTotal={'$100.00'}
        />
    )

    expect(wrapper).toMatchSnapshot()
})


test('Render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(
        <ExpensesSummary
            expensesCount={3}
            expensesTotal={'$340.60'}
        />
    )

    expect(wrapper).toMatchSnapshot()
})