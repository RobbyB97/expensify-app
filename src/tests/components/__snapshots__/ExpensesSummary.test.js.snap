// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ExpensesSummary with multiple expenses 1`] = `
<div>
  You have 3 expenses totalling $340.60.
</div>
`;

exports[`Render ExpensesSummary with no expenses 1`] = `
<div>
  You have no expenses.
</div>
`;

exports[`Render ExpensesSummary with one expense 1`] = `
<div>
  You have 1 expenses totalling $100.00.
</div>
`;
